engine/crates/runtime/src/kv.rs

```diff
@@ -1,37 +1,47 @@
-use std::time::Duration;
+use std::{sync::Arc, time::Duration};
 
 use serde::{de::DeserializeOwned, Serialize};
 
+#[derive(Debug, thiserror::Error)]
+pub enum KvError {
+    #[error("Kv error: {0}")]
+    Kv(String),
+    #[error("Serialization error: {0}")]
+    Serialization(#[from] serde_json::Error),
+}
+
+pub type KvResult<T> = Result<T, KvError>;
+
+// used in gateway-core for jwt-verifier, not sure if relevant outside.
```

Comment:
Changing the `runtime::kv` because it relied heavily on type parameters which made it painful to use behind a `dyn`